Additional AVS entries for AVS[10-15] (#330)
Azure/Azure-Proactive-Resiliency-Library: docs/content/services/specialized-workloads/azure-vmware-solution/_index.md

**Recommendation/Guidance**

> If a Multi-AZ deployment of Azure VMware Solution is required, needs an infrastructure SLA of 99.99%, or needs synchronous storage replication between AZs (RPO=0), then Azure VMware Solution Stretched Clusters should be considered. If you are in a region that supports stretched clusters, consider enabling this feature to spread the VMware vSAN datastore across two availability zones. Note: Configuring an Azure VMware Solution private cloud as a stretched cluster can only be done during initial implementation and requires twice the quota. This is due to a stretched cluster extending the cluster to the second availability zone.

Reviewer: ```suggestion
If a Multi-AZ deployment of Azure VMware Solution is required, needs a financially backed SLA of 99.99%, or needs synchronous storage replication between AZs (RPO=0), then Azure VMware Solution Stretched Clusters should be considered. If you are in a region that supports stretched clusters, consider enabling this feature to spread the VMware vSAN datastore across two availability zones. Note: Configuring an Azure VMware Solution private cloud as a stretched cluster can only be done during initial implementation and requires twice the quota. This is due to a stretched cluster extending the cluster to the second availability zone.
```

**Recommendation/Guidance**

> Alert when the cluster size of 14 hosts is reached. Periodically fire up alerts to plan for a new cluster or additional datastore if growth is driven solely by storage. Beyond 14 hosts, every time a new host is added, surface an alert.

Reviewer: ```suggestion
Alert when the cluster size of 14 hosts is reached. Additionally, periodic alerts should be set up to indicate when growth, especially driven by storage requirements, necessitates planning for a new cluster or the addition of extra datastores. Furthermore, beyond the threshold of 14 hosts, alerts should be triggered each time a new host is added to the cluster, allowing proactive monitoring and management of resource utilization.
```

{{< /collapse >}}

<br><be>

### AVS-10 - Align ExpressRoute configuration with best practices for circuit resilience

**Category: Networking**

**Impact: High**

**Recommendation/Guidance**

For critical workloads, Microsoft recommends deploying two (or more) ExpressRoute circuits in different ExpressRoute peering locations. Use Global Reach to connect multiple ExpressRoute circuits and your Azure VMware Solutions private clouds. Please review the APRL recommendations for ExpressRoute circuits in the Resources section below.

**Resources**

- [APRL guidance for ExpressRoute circuits](https://azure.github.io/Azure-Proactive-Resiliency-Library/services/networking/expressroute-circuits)
- [Create a new ExpressRoute circuit](https://learn.microsoft.com/azure/expressroute/expressroute-howto-circuit-portal-resource-manager?pivots=expressroute-preview#create-a-new-expressroute-circuit-preview)

**Resource Graph Query/Scripts**

{{< collapse title="Show/Hide Query/Script" >}}

{{< code lang="sql" file="code/avs-10/avs-10.kql" >}} {{< /code >}}

{{< /collapse >}}

> <br><be>

Reviewer: ```suggestion
<br><br>
```

### AVS-11 - Deploy two or more circuits in different peering locations when using stretched clusters

**Category: Networking**

**Impact: High**

**Recommendation/Guidance**

Azure VMware Solution vSAN stretched clusters span two Availability Zones (AZs) in the region where they are deployed (plus a third AZ for the witness node). When using ExpressRoute to connect to the vSAN stretched clusters from  on-premises, align the ExpressRoute implementation's resilience to the clusters’ resilience by deploying two circuits in different peering locations (i.e., different sites/DC facilities). When using Global Reach, implement a  mesh topology by connecting the on-premises circuits to the managed circuits provided by the Azure VMware Solution private cloud.

**Resources**

- [Deploy vSAN streched cluster](https://learn.microsoft.com/en-us/azure/azure-vmware/deploy-vsan-stretched-clusters#deploy-a-stretched-cluster-private-cloud)

**Resource Graph Query/Scripts**

{{< collapse title="Show/Hide Query/Script" >}}

{{< code lang="sql" file="code/avs-11/avs-11.kql" >}} {{< /code >}}

{{< /collapse >}}

> <br><be>

Reviewer: ```suggestion
<br><br>
```

### AVS-12 - Deploy two Azure VMware Solution private clouds in different regions for geographical disaster recovery

**Category: Disaster Recovery**

**Impact: High**

**Recommendation/Guidance**

Two Azure VMware Solution private clouds can be deployed in different regions for business continuity. Implement a  mesh network topology based on ExpressRoute Gateway Connections and Global Reach Connections.

**Resources**

- [Private Clouds in two regions](https://learn.microsoft.com/en-us/azure/azure-vmware/move-azure-vmware-solution-across-regions)
- [Dual Region Network Topology](https://learn.microsoft.com/en-us/azure/cloud-adoption-framework/scenarios/azure-vmware/eslz-dual-region-network-topology)

**Resource Graph Query/Scripts**

{{< collapse title="Show/Hide Query/Script" >}}

{{< code lang="sql" file="code/avs-12/avs-12.kql" >}} {{< /code >}}

{{< /collapse >}}

> <br><be>

Reviewer: ```suggestion
<br><br>
```

**Resources**

- [Connect Private Clouds in the same region](https://learn.microsoft.com/en-us/azure/azure-vmware/connect-multiple-private-clouds-same-region)

**Resource Graph Query/Scripts**

{{< collapse title="Show/Hide Query/Script" >}}

{{< code lang="sql" file="code/avs-13/avs-13.kql" >}} {{< /code >}}

{{< /collapse >}}

<br><br>

### AVS-14 - Use key autorotation for vSAN datastore customer-managed keys

**Category: Storage**

**Impact: High**

**Recommendation/Guidance**

When using customer-managed keys to encrypt the vSAN datastore(s), use Azure Key Vault for centralized management and access them using a managed identity mapped to the private cloud. Key expiration can result in the vSAN datastore and its workloads becoming unavailable. Configure key autorotation to avoid unplanned outages due to key rotation not occurring before expiration.

**Resources**

- [Configure Customer Managed Keys](https://learn.microsoft.com/en-us/azure/azure-vmware/configure-customer-managed-keys?tabs=azure-portal)

**Resource Graph Query/Scripts**

{{< collapse title="Show/Hide Query/Script" >}}

{{< code lang="sql" file="code/avs-14/avs-14.kql" >}} {{< /code >}}

{{< /collapse >}}

> <br><be>

Reviewer: ```suggestion
<br><br>
```
